src/App.jsx

```diff
@@ -1,23 +1,27 @@
 import React, { useState, useEffect } from 'react';
+import { ColorRing } from 'react-loader-spinner';
 import data from './data.json';
 
 import Header from './Header';
 import ToDoList from './ToDoList';
 import ToDoForm from './ToDoForm';
 
+async function fetchData(setToDoList) {
```

Comment:
i meant move it outside of useEffect, but keep it inside the component(:
let's move this inside the component. you may use `useCallback` hook (read about it and let me know if you need more explanation about it)

in general it is not recommended to overuse useCallbacks (some suggest not to use it at all), but we are doing it for the sake of getting to know React better